orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java

```diff
@@ -897,4 +897,22 @@ private SaltConnector createSaltConnector(GatewayConfig gatewayConfig) {
             throw new RuntimeException("Cannot determine service endpoint for gateway: " + gatewayConfig, e);
         }
     }
+
+    @Override
+    public void uploadIdpMetadataXml(List<GatewayConfig> allGatewayConfigs, ExitCriteriaModel exitModel, String metadataXml)
+            throws CloudbreakOrchestratorFailedException {
+        GatewayConfig gateway = getPrimaryGatewayConfig(allGatewayConfigs);
+        Set<String> gatewayTargets = getGatewayPrivateIps(allGatewayConfigs);
+        try (SaltConnector saltConnector = createSaltConnector(gateway)) {
+            if (!gatewayTargets.isEmpty()) {
+                LOGGER.debug("Upload metadata.xml to gateways");
+                byte[] metadataXmlBytes = metadataXml.getBytes();
+                String samlPath = "/var/lib/knox/cloudbreak_resources/";
```

Comment:
Silly question: How will this get moved to `/etc/knox/sso/saml/metadata.xml` (as expected by the SSO topology)? Is not that something to be done in `orchestrator-salt/src/main/resources/salt/salt/gateway/cm.sls`?

Reply:
Good catch! This is a mistake from my part. I modified this part after testing the feature, but forgot to sync it with the path in knoxsso.xml